core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/GuardrailResult.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+import java.util.List;
+
+public interface GuardrailResult<GR extends GuardrailResult> {
+    enum Result {
+        SUCCESS,
+        FAILURE,
+        FATAL
+    }
+
+    boolean isSuccess();
+
+    boolean isFatal();
+
+    List<? extends Failure> getFailures();
+
+    Throwable getFirstFailureException();
+
+    GR validatedBy(Class<? extends Guardrail> guardrailClass);
+
+    class Failure {
+        private final String message;
+        private final Throwable cause;
+
+        private Class<? extends Guardrail> guardrailClass;
```

Comment:
should be guarded.

Reply:
> should be guarded

Sorry, but I don't understand what you mean, can you clarify?

Reply:
it can be written and read concurrently. So, we either make it final (with a copy on write implementation), or make sure there is concurrent access (not to the list itself, but to the field)

Reply:
I structurally solved the problem turning all classes belonging to the API into records.